home/models.py

```diff
@@ -1215,3 +1216,285 @@ class PageView(models.Model):
+        to="home.WhatsAppTemplate",
+        on_delete=models.CASCADE,
+        related_name="template_quick_reply_items",
+    )
+
+
+class WhatsAppTemplate(
+    DraftStateMixin, ClusterableModel, RevisionMixin, index.Indexed, models.Model
+):
+
+    class Category(models.TextChoices):
+        UTILITY = "UTILITY", _("Utility")
+        MARKETING = "MARKETING", _("Marketing")
+
+    name = models.CharField(max_length=512, blank=True, default="")
+    category = models.CharField(
+        max_length=14,
+        choices=Category.choices,
+        default=Category.MARKETING,
+    )
+    quick_replies = ClusterTaggableManager(
+        through="home.TemplateQuickReplyContent", blank=True
+    )
+
+    related_pages = StreamField(
+        [
+            ("related_page", blocks.PageChooserBlock()),
+        ],
+        blank=True,
+        null=True,
+        use_json_field=True,
+    )
+
+    locale = models.ForeignKey(Locale, on_delete=models.CASCADE, default="")
+
+    image = models.ForeignKey(
+        "wagtailimages.Image",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="image",
+    )
+    message = models.TextField(
+        help_text="each text message cannot exceed 4096 characters, messages with "
+        "media cannot exceed 1024 characters.",
+        max_length=4096,
+    )
+
+    example_values = StreamField(
+        [
+            (
+                "example_values",
+                blocks.CharBlock(
+                    label="Example Value",
+                ),
+            )
+        ],
+        blank=True,
+        null=True,
+        use_json_field=True,
+    )
+
+    search_fields = [
+        index.SearchField("locale"),
+    ]
+
+    @property
+    def fields(self):
+        """
+        Returns a tuple of fields that can be used to determine template equality
+        """
+        return (
+            self.message,
+            self.image,
+            self.category,
+        )
+
+    @property
+    def quick_reply_buttons(self):
+        return self.template_quick_reply_items.all().values_list("tag__name", flat=True)
+
+    @property
+    def prefix(self):
+        return self.name.lower().replace(" ", "_")
+
+    def status(self):
+        return "Live" if self.live else "Draft"
+
+    # # TODO: Figure out which of these needs to call the update
+    # We need to relook at the states of a standalone template, its not just draft and live, theres also pending, declined etc
+    # def save(self, *args, **kwargs):
+    #     if kwargs.get("update_fields"):
+    #         # save not called from publish
+    #         # do_stuff_on_save()
+    #         print("THIS IS A SAVE")
+    #     else:
+    #         # do_stuff_on_publish()
+    #         print("THIS IS A PUBLISH")
+    #     return super().save(*args, **kwargs)
+
+    def __str__(self):
+        """String repr of this snippet."""
+        return self.name
+
+    class Meta:  # noqa
```

Comment:
Should `Meta` be closer to the top of the class?

Reply:
Yup, moved